### ISSUE TITLE
Darwin - Missing support

### ISSUE THREAD

**Issue by noelmcloughlin** [saltstack-formulas/tomcat-formula#70]
Need support for Darwin (MacOS).

```
[ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'main_config_template'
[CRITICAL] Rendering SLS 'base:tomcat.config' failed: Jinja variable 'dict object' has no attribute 'main_config_template'
[ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'native_pkg'
[CRITICAL] Rendering SLS 'base:tomcat.native' failed: Jinja variable 'dict object' has no attribute 'native_pkg'
[ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'manager_pkg'
[CRITICAL] Rendering SLS 'base:tomcat.manager' failed: Jinja variable 'dict object' has no attribute 'manager_pkg'
local:
    Data failed to compile:
----------
    Rendering SLS 'base:tomcat.config' failed: Jinja variable 'dict object' has no attribute 'main_config_template'
----------
    Rendering SLS 'base:tomcat.native' failed: Jinja variable 'dict object' has no attribute 'native_pkg'
----------
    Rendering SLS 'base:tomcat.manager' failed: Jinja variable 'dict object' has no attribute 'manager_pkg'
```

**Comment by noelmcloughlin** [comment]
See saltstack-formulas/postgres-formula/pull/164 for reference solution.

**Comment by noelmcloughlin** [comment]
Port exists.
```
$ brew install tomcat
==> Downloading https://www.apache.org/dyn/closer.cgi?path=tomcat/tomcat-8/v8.5.21/bin/apache-tomcat-8.5.21.tar.gz
==> Best Mirror http://mirrors.whoishostingthis.com/apache/tomcat/tomcat-8/v8.5.21/bin/apache-tomcat-8.5.21.tar.gz
######################################################################## 100.0%
==> Caveats
To have launchd start tomcat now and restart at login:
  brew services start tomcat
Or, if you don't want/need a background service you can just run:
  catalina run
==> Summary
🍺  /usr/local/Cellar/tomcat/8.5.21: 632 files, 12.8MB, built in 39 seconds
C1IRE-LM22900:tomcat nmcloughlin$ brew link tomcat
Warning: Already linked: /usr/local/Cellar/tomcat/8.5.21
To relink: brew unlink tomcat && brew link tomcat
```

**Comment by noelmcloughlin** [comment]
Refer to saltstack-formulas/postgres-formula/issues/162

**Comment by noelmcloughlin** [comment]
I got tomcat working on Darwin:  [tomcat_on_Darwin_verified.txt](https://github.com/saltstack-formulas/tomcat-formula/files/1356212/tomcat_on_Darwin_verified.txt)

**Comment by noelmcloughlin** [comment]
Just after salt-formula completes launchd keeps retrying tomcat startup as uid 0. This fails so launchd gives up (Address already in use).  Tomcat is running as unprivledged user.

This means formula works for Darwin.  Tested reboot and launchd starts tomcat as un-privledged user as expected (LaunchAgents start with permissions of currently logged users).

**Comment by noelmcloughlin** [comment]
Preparing one PR for each tomcat.state and one PR for pillar/os/defaults.

**Comment by noelmcloughlin** [comment]
Requires Homebrew (https://brew.sh/) and Homebrew services (https://github.com/Homebrew/homebrew-services)

**Comment by noelmcloughlin** [comment]
I have some ideas why Salt/Launchd spawns two process starts during salt-formula execution on MacOS/Darwin. It might be related decision to integrate with /Library/LaunchAgents/ and we have two sessions (privileged sudo, unprivileged logged on user) open on Darwin so launchd decides both sessions need Tomcat started.   To figure this out is another days work ;-)

**Comment by noelmcloughlin** [comment]
MacOS support was merged by #71, #72, #73, #74, #75, #76.  Improvement needed: #77
